src/auto/injector.js

```diff
@@ -938,14 +938,15 @@ function createInjector(modulesToLoad, strictDi) {
         fn = fn[fn.length - 1];
       }
 
-      if (!isClass(fn)) {
-        // http://jsperf.com/angularjs-invoke-apply-vs-switch
-        // #5388
-        return fn.apply(self, args);
-      } else {
+      if (isClass(fn)) {
+        // This code path was used by $controller previously and is not really needed any more.
```

Comment:
Can we remove this path? Or would it be considered a breaking change in the public API? This behavior of `invoke` isn't documented.

Reply:
Since `fn` is documented to be a function (not a class) and it doesn't make sense to call invoke on a class, it is fine to remove this imo.